### PR TITLE
Update auth and http client

### PR DESCRIPTION
**Pull request by elharo**
@suztomo @chingor13 I had to swap the order of dependencies we declare, though this really shouldn't matter.

### CODE REVIEW

File: boms/cloud-oss-bom/pom.xml [hunk 1]
```diff
@@ -67,6 +67,7 @@
 
   <dependencyManagement>
     <dependencies>
+
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:70]: Unnecessary empty line.

**Review by suztomo — APPROVED** [review]
Nice solution.

File: boms/cloud-oss-bom/pom.xml (hunk 1 above)
Comment (elharo) [reply boms/cloud-oss-bom/pom.xml:70]: removed

**Review by elharo — COMMENTED** [review]
I'm still not happy with this. It unblocks us, but ideally the upper bounds conditions should be satisfied in any order.

**Pull request merged**